### DISCUSSION TITLE
Is it possible to use Vue 3 dynamic component on element-plus?

### DISCUSSION THREAD

**Discussion by zeshhaan** [element-plus/element-plus#6326] [Q&A] [answered]
Hi!
I am looking for a way to switch between el-button and el-link using Vue 3's [dynamic component](https://vuejs.org/guide/essentials/component-basics.html#dynamic-components) feature. When i tried, i couldn't get it to work. Is it possible to get it working? Or can anyone point out what i would be potentially missing here?

[Issue reproduction](https://stackblitz.com/edit/vitejs-vite-akdkhy?file=src/App.vue)

Thanks in advance 🙏

**Comment by msidolphin** [comment, marked as answer]
If you return string you must have registered the component, otherwise you can just return the component

```vue
<template>
  <component :is="tagName">
    <slot />
  </component>
</template>

<script setup lang="ts">
import { computed } from 'vue'
import { ElButton, ElLink } from '@element-plus/components'
const props = defineProps({
  href: {
    type: String,
    required: false,
    default: null,
  },
})
const tagName = computed(() => {
  if (props.href) {
    return ElLink
  } else return ElButton
})
</script>
```
